code/obj/item/swingsign.dm

```diff
@@ -0,0 +1,197 @@
+#define CHARACTERLIMIT 350
+
+//1. /obj/swingsign
+//	Inherited procs
+//	tgui procs
+//	procs
+//2. /obj/item/swingsignfolded
+
+TYPEINFO(/obj/swingsign)//No idea what TYPEINFO is, I just know it lets me disable the brown overcoat when crafting signs
+	mat_appearances_to_ignore = list("wood")
+/obj/swingsign
+	name = "swing sign"
+	desc = "A foldable sign for writing annoucements or advertisements."
+	icon = 'icons/obj/furniture/swingsign.dmi'
+	icon_state = "blank"
+	throwforce = 10
+	density = 1
+	custom_suicide = 1
+	/// Stored message
+	var/message = ""
+	var/defaultdesc = "A foldable sign for writing annoucements or advertisements."
+	/// Text inserted before the actual message
+	var/descpreamble = "It says:<br><div style='text-align:center'>"// There's a /div in SetMessage. If changing the preamble remember to respect it.
+	var/secured = FALSE
+	/// Damage when thrown into a swing sign
+	var/maxcrashdamage = 5
+	/// Max length of the message
+	var/maxmessagerows = 10
+	/// Max width of the message
+	var/maxmessagecols = 40
+	/// Regex that detects '\n' signs
+	//var/static/regex/fixBreaklinesRegex = new(@{"/\n/"}, "g")
+	/// Regex that detects '<br>' tags
+	//var/static/regex/fixBreaklinesRegex = new(@{"<br>"}, "g")
+
+	New()
+		..()
+		if(message)
+			setmessage(message)
+
+	proc/setmessage(var/newmessage)
+		message = newmessage
+		if(message == "")
+			desc = defaultdesc
+		else
+			desc = descpreamble + message + "</div>"
+		UpdateIcon()
+
+	update_icon(...)
+		if(message == "")
+			icon_state = "blank"
+		else
+			icon_state = "written"
+
+	proc/fold()
+		var/obj/item/swingsignfolded/newSwingsign = new/obj/item/swingsignfolded(src.loc)
+
+		if (src.material)
+			newSwingsign.setMaterial(src.material)
+		//if (src.icon_state)
+		//	newSwingsign.icon_state = "folded"
+		if (src.message)
+			newSwingsign.message = src.message
+
+		qdel(src)
+
+	attack_hand(mob/user)
+		if (!isliving(user)) return
+		if(!secured)
+			user.visible_message("<b>[user.name] folds [src].</b>")
+			playsound(src, 'sound/impact_sounds/Clock_slap.ogg',50,1)
+			fold()
+		else
+			boutput(user, "<span alert='notice'>Sign is too tightly secured to fold!</span>")
+		return
+
+	attackby(obj/item/W, mob/user)
+		if(istype(W, /obj/item/pen))
+			ui_interact(user)
+			return
+		else if (isscrewingtool(W))
+			if(secured)
+				boutput(user, "<span class='notice'>You unsecure the swing sign.</span>")
+				anchored = UNANCHORED
+			else
+				boutput(user, "<span class='notice'>You secure the swing sign.</span>")
+				anchored = ANCHORED
+			playsound(src, 'sound/items/Screwdriver.ogg', 50, 1)
+			secured = !secured
```

Comment:
do you really need this var? Seems like you could just rely on anchored directly